spec/rails_helper.rb

```diff
@@ -68,4 +68,5 @@
   config.include Devise::Test::ControllerHelpers, type: :view
   # https://github.com/heartcombo/devise#integration-tests
   config.include Devise::Test::IntegrationHelpers, type: :feature
+  config.include Devise::Test::IntegrationHelpers, type: :request
```

Comment:
What does this do?

Reply:
Does this help? https://github.com/heartcombo/devise/wiki/How-To:-sign-in-and-out-a-user-in-Request-type-specs-(specs-tagged-with-type:-:request)

Reply:
The line is necessary to access the devise test helpers (e.g. `sign_in`) in request specs.